src/main/java/picard/sam/DownsampleSam.java

```diff
@@ -115,10 +133,12 @@ protected int doWork() {
         final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(in.getFileHeader(), true, OUTPUT);
         final ProgressLogger progress = new ProgressLogger(log, (int) 1e7, "Wrote");
         final DownsamplingIterator iterator = DownsamplingIteratorFactory.make(in, STRATEGY, PROBABILITY, ACCURACY, RANDOM_SEED);
+        final PublicCollectQualityYieldMetrics metricsCollector = new PublicCollectQualityYieldMetrics();
 
         while (iterator.hasNext()) {
             final SAMRecord rec = iterator.next();
             out.addAlignment(rec);
+            metricsCollector.acceptRead(rec, null);
```

Comment:
this should only be done if METRIC_FILE !=null since it needs a more thorough parsing of the bam-file than downsampling...